zhaquirks/ikea/symfonisk.py

```diff
@@ -196,3 +212,131 @@ class IkeaSYMFONISK2(CustomDevice):
+        # device_version=1
+        # input_clusters=[0, 1, 3, 32, 4096, 64599]
+        # output_clusters=[3, 4, 6, 8, 25, 4096, 64639]>
+        MODELS_INFO: [(IKEA, "SYMFONISK sound remote gen2")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.REMOTE_CONTROL,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration.cluster_id,
+                    Identify.cluster_id,
+                    PollControl.cluster_id,
+                    LightLink.cluster_id,
+                    WWAH_CLUSTER_ID,
+                ],
+                OUTPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    LevelControl.cluster_id,
+                    Ota.cluster_id,
+                    LightLink.cluster_id,
+                    V1ShortcutCluster.cluster_id,
+                ],
+            }
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.REMOTE_CONTROL,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration2AAACluster,
+                    Identify.cluster_id,
+                    PollControl.cluster_id,
+                    LightLink.cluster_id,
+                    WWAH_CLUSTER_ID,
+                ],
+                OUTPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    LevelControlCluster,
+                    Ota.cluster_id,
+                    LightLink.cluster_id,
+                    V1ShortcutCluster,
+                ],
+            }
+        }
+    }
+
+    device_automation_triggers = {
+        (SHORT_PRESS, TURN_ON): {
+            COMMAND: COMMAND_TOGGLE,
+        },
+        (SHORT_PRESS, DIM_UP): {
+            COMMAND: COMMAND_MOVE_ON_OFF,
+            PARAMS: {"move_mode": 0},
+        },
+        (SHORT_PRESS, DIM_DOWN): {
+            COMMAND: COMMAND_MOVE_ON_OFF,
+            PARAMS: {"move_mode": 1},
+        },
+        (LONG_PRESS, DIM_UP): {
+            COMMAND: COMMAND_MOVE,
+            PARAMS: {"move_mode": 0},
+        },
+        (LONG_PRESS, DIM_DOWN): {
+            COMMAND: COMMAND_MOVE,
+            PARAMS: {"move_mode": 1},
+        },
+        (SHORT_PRESS, LEFT): {
+            COMMAND: COMMAND_STEP,
+            CLUSTER_ID: 8,
+            PARAMS: {"step_mode": 1},
+        },
+        (SHORT_PRESS, RIGHT): {
+            COMMAND: COMMAND_STEP,
+            CLUSTER_ID: 8,
+            PARAMS: {"step_mode": 0},
+        },
+        (SHORT_PRESS, BUTTON_1): {
+            COMMAND: COMMAND_STEP_ON_OFF,
+            CLUSTER_ID: 8,
+            "args": {"step_mode": 0, "step_size": 1},
```

Comment:
Replace all these `"args"` with `PARAMS` like the previous ones

Reply:
The problem with this is that only args is populated, but not params. If I change it to PARAMS, both shortcut buttons will fire both events.

This is also visible in the zha_event:
```
event_type: zha_event
data:
  device_ieee: [...]
  unique_id: [...]
  device_id: [...]
  endpoint_id: 1
  cluster_id: 8
  command: step_with_on_off
  args:
    step_mode: 0
    step_size: 1
    transition_time: 0
  params: {}
origin: LOCAL
time_fired: "2023-02-28T22:05:34.938744+00:00"
context:
  id: [...]
  parent_id: null
  user_id: null
```

I'm currently trying to figure out how to both populate args and params. If you know what the problem could be, I'd be happy about some tips.

Reply:
The system is not knowing wot the args is meaning then we have not defined them.
I wonder is its right way using it on level cluster or crafting one new type and translating  the args to parms like single double and long press like is made for the scenes on the five button remote.

Reply:
Maybe is a nosense but that reminds me to the `level` attribute in the MCU Tuya dimmers:
https://github.com/zigpy/zha-device-handlers/blob/1d39b0588435adda64df1357285f4ecddd78e656/zhaquirks/tuya/mcu/__init__.py#L508-L516

Can it be that when using the `PARAMS` the values came in the `*,` part?
```python
    def handle_cluster_request(
        self,
        hdr: foundation.ZCLHeader,
        args: List[Any],
        *,
        dst_addressing: Optional[
            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
        ] = None,
    ):
```

(Sorry if this is a silly asumption but I'm not fluent in Python)

Reply:
@javicalle Tbh you pretty much got me to the solution! It might not have been these exact lines, but a few lines below they send a t.Struct TuyaClusterData to listener_event. Then I again looked at how the method listener_event works and noticed that I've been interpreting an if clause wrong. So... thanks!